event_response_tool/event_response.py

```diff
@@ -0,0 +1,323 @@
+import os
+import warnings
+import logging
+import logging.config
+import shutil
+import configparser
+import argparse
+from datetime import datetime
+from types import SimpleNamespace
+
+import analyzere
+
+from utils.alert import Alert as alert
+from utils.file_handler import read_input_file, file_exists
+from layer_loss_duplicator.duplicate_layer_loss import LayerLossDuplicator
+from ap_creator.create_ap import AnalysisProfileCreator
+
+
+# Load config file
+config_file = os.path.abspath("config/event_response_config.ini")
+if not os.path.exists(config_file):
+    alert.error("Config file not present. Cannot continue.")
+config = configparser.ConfigParser()
+config.read(config_file)
+
+
+class EventResponseHandler:
+    def __init__(self, output_dir, event_response_inputs):
+        self.output_dir = output_dir
+        self.event_response_inputs = event_response_inputs
+
+    def execute(self):
+        self.event_weights_df = read_input_file(
+            self.event_response_inputs.event_weights_csv
+        )
+        if (
+            self.event_response_inputs.layer_views_csv
+            or self.event_response_inputs.portfolio_view_uuid
+        ):
+            modify_layer_loss = LayerLossDuplicator(
+                config=config,
+                output_dir=self.output_dir,
+                event_weights_df=self.event_weights_df,
+                layer_ids_csv=self.event_response_inputs.layer_views_csv,
+                portfolio_uuid=self.event_response_inputs.portfolio_view_uuid,
+            )
+            modify_layer_loss.modify_layer_loss_data()
+        else:
+            ap_creator = AnalysisProfileCreator(
+                config=config,
+                event_weights_df=self.event_weights_df,
+                total_num_of_events=self.event_response_inputs.total_number_of_events,
+                trial_count=self.event_response_inputs.trial_count,
+                catalog_description=self.event_response_inputs.catalog_description,
+                simulation_description=self.event_response_inputs.simulation_description,
+                analysis_profile_description=self.event_response_inputs.analysis_profile_description,
+                old_analysis_profile_uuid=self.event_response_inputs.old_analysis_profile_uuid,
+            )
+            if self.event_response_inputs.old_analysis_profile_uuid:
```

Comment:
Given that this is passed to AnalysisProfileCreator, I this it makes more sense to either put this `if` inside the class, or pass the old AP uuid to the update function.